### PR TITLE
prepare for v1

### PR DESCRIPTION
**Pull request by markandersontrocme**
### Description of your changes

- Move to TS
- use api-token
- get latest current version if cached
- small fixes

Fixes https://github.com/upbound/action-up/issues/5

I have:

- [ ] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -53,9 +53,49 @@ jobs:
...
+
+      - name: Install up
+        id: test-action
+        uses: ./
+        with:
+          version: v0.32.0
+          skip-login: true
+
+      - name: Verify Installation
+        run: up version
```
Comment (adamwg) [inline .github/workflows/ci.yml:101]: Can we validate the output here to make sure we got the right version?

File: src/main.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
+import * as core from '@actions/core'
+import * as exec from '@actions/exec'
+import * as tc from '@actions/tool-cache'
+import * as fs from 'fs'
+
+const upToolname = 'up'
+const currentVersionUrl = 'https://cli.upbound.io/stable/current/version'
```
Comment (adamwg) [inline src/main.ts:7]: This should interpolate the `channel` in place of `stable` so that we get the right current version on non-stable channels.

**Review by adamwg — CHANGES_REQUESTED** [review]
One thing we should change in the current version check - otherwise, lgtm.

File: src/main.ts (hunk 2 above)
Comment (markandersontrocme) [reply src/main.ts:7]: ah right good catch

**Review by adamwg — APPROVED** [review]
lgtm!

**Pull request merged**